gems/quilt_rails/lib/quilt_rails/trusted_ui_server_csrf_strategy.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+module Quilt
+  class TrustedUiServerCsrfStrategy
+    def initialize(controller)
+      @controller = controller
+    end
+
+    def handle_unverified_request
+      return if node_server_side_render?
+
+      null_session.handle_unverified_request
+    end
+
+    private
+
+    def node_server_side_render?
+      @controller.request.headers['x-shopify-server-side-rendered'] == '1'
+    end
+
+    def null_session
```

Comment:
🤔 do we want to handle non-header requests via the Exception or NullSession strategy? Seems like currently we're using Exception but naming it null_session.

Reply:
Whoops.  Definitely `Exception`.  I'll update 👍